bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/PatientDaoImpl.java

```diff
@@ -88,18 +87,70 @@ public PatientDaoImpl(SessionFactory sessionFactory) {
+                    return toPatientResponse(patientResponseMapper, patient, loginLocationUuid, addressSearchResultFields, patientSearchResultFields, programAttributes, uniquePatientIds);
                 }).filter(Objects::nonNull)
                 .collect(toList());
         return patientResponses;
     }
 
+    @Override
+    public List<PatientResponse> getSimilarPatientsUsingLuceneSearch(String name, String gender, String loginLocationUuid, Integer length) {
+        PatientResponseMapper patientResponseMapper = new PatientResponseMapper(Context.getVisitService(),new BahmniVisitLocationServiceImpl(Context.getLocationService()));
+        List<Patient> patients = getPatientsByNameAndGender(name, gender, length);
+        Set<Integer> uniquePatientIds = new HashSet<>();
+        // TODO BAH-460 Maybe we can remove the new HashMap<>() from the call. It used to be response from validateSearchParams(...)
+        List<PatientResponse> patientResponses = patients.stream()
+                .map(patient -> toPatientResponse(patientResponseMapper, patient, loginLocationUuid, new HashMap<>(), uniquePatientIds)).filter(Objects::nonNull)
+                .collect(toList());
+        return patientResponses;
+    }
+
+    private PatientResponse toPatientResponse(PatientResponseMapper patientResponseMapper, Patient patient, String loginLocationUuid, Map<Object, Object> programAttributes, Set<Integer> uniquePatientIds) {
+        return toPatientResponse(patientResponseMapper, patient, loginLocationUuid, null, null, programAttributes, uniquePatientIds);
+    }
+
+    private PatientResponse toPatientResponse(PatientResponseMapper patientResponseMapper, Patient patient, String loginLocationUuid, String[] addressSearchResultFields, String[] patientSearchResultFields, Map<Object, Object> programAttributes, Set<Integer> uniquePatientIds) {
+        if(!uniquePatientIds.contains(patient.getPatientId())) {
+            PatientResponse patientResponse = patientResponseMapper.map(patient, loginLocationUuid, patientSearchResultFields, addressSearchResultFields,
+                    programAttributes.get(patient.getPatientId()));
+            uniquePatientIds.add(patient.getPatientId());
+            return patientResponse;
+        } else {
+            return null;
+        }
+    }
+
+    private List<Patient> getPatientsByNameAndGender(String name, String gender, Integer length) {
+        HibernatePatientDAO patientDAO = new HibernatePatientDAO();
+        patientDAO.setSessionFactory(sessionFactory);
+        List<Patient> patients = new ArrayList<Patient>();
+        String query = LuceneQuery.escapeQuery(name);
+        PersonLuceneQuery personLuceneQuery = new PersonLuceneQuery(sessionFactory);
+        LuceneQuery<PersonName> nameQuery = personLuceneQuery.getPatientNameQueryWithOrParser(query, false);
+        /* person.gender does not work somehow in LuceneQuery, so the dirty way is to filter result with person's gender */
```

Comment:
please remove this comment, we should explain in the JIRA ticket that we decided to go this route as to not have to adapt the openmrs lucene index (since this is where gender would need to be included in the index)